src/dswx_sar/save_mgrs_tiles.py

```diff
@@ -88,41 +127,179 @@ def save_mgrs_tile(source_tif_path,
     xspacing = input_tif_obj.GetGeoTransform()[1]
     yspacing = input_tif_obj.GetGeoTransform()[5]
 
-    # extract epsg from input
-    proj = osr.SpatialReference(wkt=input_tif_obj.GetProjection())
-    epsg_code_input = proj.GetAttrValue('AUTHORITY', 1)
-
     # compute EPSG for given mgrs_tile_id
-    mgrs_obj = mgrs.MGRS()
-    utm_zone = mgrs_obj.MGRSToUTM(output_mgrs_id)
+    # Compute bounding extents for given MGRS tile
 
-    if utm_zone[1] == 'N':
-        is_southern = False
-    else:
-        is_southern = True
-    crs = CRS.from_dict({'proj': 'utm', 'zone': utm_zone[0], 'south': is_southern})
-    epsg_output = crs.to_authority()[1]
+    x_value_min, x_value_max, y_value_min, y_value_max, epsg_output = \
+        get_geographic_boundaries_from_mgrs_tile(output_mgrs_id,
+                                                 mgrs_db_path,)
 
-    # Compute bounding extents for given MGRS tile
-    x_value_min, x_value_max, y_value_min, y_value_max = \
-        get_geographic_boundaries_from_mgrs_tile(output_mgrs_id)
     bbox = [x_value_min, y_value_min, x_value_max, y_value_max]
 
     # create MGRS tiles for products.
     output_tif_file_path = f'{output_dir_path}/{output_tif_name}'
-    ds = input_tif_obj
     opt = gdal.WarpOptions(dstSRS=f'EPSG:{epsg_output}',
-                     outputType=band.DataType,
-                     xRes=xspacing,
-                     yRes=yspacing,
-                     outputBounds=bbox,
-                     resampleAlg=method,
-                     dstNodata=no_data_value,
-                     format='GTIFF')
+                           outputType=band.DataType,
+                           xRes=xspacing,
+                           yRes=yspacing,
+                           outputBounds=bbox,
+                           resampleAlg=method,
+                           format='GTIFF')
     gdal.Warp(output_tif_file_path, source_tif_path, options=opt)
-    dswx_sar_util._save_as_cog(output_tif_file_path, output_dir_path, logger,
-                compression='ZSTD',
-                nbits=16)
+    input_tif_obj = None
+    dswx_sar_util._save_as_cog(output_tif_file_path,
+                               output_dir_path,
+                               logger,
+                               compression='ZSTD',
+                               nbits=16)
```

Comment:
You probably want to expose these parameters. For RTC-S1, we noticed that the `ZSTD` compression is not currently handled by the Google Earth Engine, and so we changed it to `DEFLATE`. Same for the number of bits. We had to change it back to 32.